### PR TITLE
Ported the code from the demo

### PR DESCRIPTION
**Pull request by NSeydoux**
This reuses the code from the original demo, with no additional
improvements.

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [X] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: cli/cli.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
```
Comment (Vinnl) [inline cli/cli.sh:1]: Shouldn't the file extension here be `cli.js`?

File: package.json [hunk 2]
```diff
@@ -3,8 +3,11 @@
   "version": "0.0.0",
   "description": "A small app to help scripts access private resources on Solid Pods",
   "main": "dist/index.js",
+  "bin": {
+    "generate-oidc-token": "cli/cli.sh"
+  },
```
Comment (Vinnl) [inline package.json:8]: Knowing ourselves, might be best to keep this stable against future package renames :P

```suggestion
  "bin": "cli/cli.sh",
```

File: package.json [hunk 3]
```diff
@@ -47,5 +52,10 @@
     "*.{js,jsx,ts,tsx}": "eslint --cache --fix",
     "*.{ts,tsx,js,jsx,css,md,mdx}": "prettier --write"
   },
-  "dependencies": {}
+  "dependencies": {
+    "@inrupt/solid-client-authn-core": "^1.2.4",
```
Comment (Vinnl) [inline package.json:56]: I don't think this would ever need to be a direct dependency for consumers?

File: src/index.ts [hunk 4]
```diff
@@ -19,6 +19,74 @@
...
+      },
+    })
+    .catch((e) => {
+      throw new Error(`Login failed: ${e.toString()}`);
+    });
+});
+
+app.get("/", async (_req, res) => {
+  console.log("Login successful.");
+  await session.handleIncomingRedirect(`${iriBase}${_req.url}`);
```
Comment (Vinnl) [inline src/index.ts:71]: I know it's kinda mandated by the common ESLint rules, but I'm just going to rage about it anyway: I don't think at all that `${iriBase}${_req.url}` is more easily readable than `iriBase + _req.url` :P

**Review by Vinnl — APPROVED** [review]

File: cli/cli.sh (hunk 1 above)
Comment (NSeydoux) [reply cli/cli.sh]: Good point. Fixed in cc1cf7bfd605100e8606b6b67f1ae5d727ce284a

File: package.json (hunk 2 above)
Comment (NSeydoux) [reply package.json]: Done in 9b35c7f0f6fe756ac6d5b43684683b8ee63826a2. I'll see if that doesn't confuse `npx` though :)

File: package.json (hunk 3 above)
Comment (NSeydoux) [reply package.json]: Excellent point, I forgot that was the case. Fixed in 71d335687c7f9606fc31c295663262ce4d72a113

File: package.json (hunk 2 above)
Comment (NSeydoux) [reply package.json]: All clear, I learned something today :)

**Pull request merged**